# The Latency Comparision (Homogeneity Test)

### Objective:
Simulate 1,000 server latencies for 2 cloud centers (in this example : Frankfurt & Munich), compare them to check if they have homogeneous variance, and observe how the Levene's Test reacts.

>Test Used: Levene's Test for Homogeneity of Variance

>Performed by : Raghav Katyal on 29 September 2026

In [1]:
#importing required libraries for the test
import numpy as np
from scipy import stats

In [2]:
#fixing seed so that values stay same
np.random.seed(42)

In [3]:
#Creating Latency Records of Cloud Center - Frankfurt
#Average is 45ms with very less standard deviation (Variance)
latency_frankfurt = np.random.normal(loc=45,scale=2,size=1000)

In [4]:
#Creating Latency Records of Cloud Center - Munich
#Average is 45ms with relatively very high standard deviation (Variance)
latency_munich = np.random.normal(loc=45,scale=15,size=1000)

In [5]:
#Running Levene's Test for Checking Homogeneity of Variance
#Expected Result : It should fail as the standard deviations of both are widely apart.

w_value,p_value = stats.levene(latency_frankfurt,latency_munich)
print(f"The W-statistic : {w_value:.3f}")
print(f"The p-value : {p_value:.3f}")


The W-statistic : 1251.603
The p-value : 0.000


>Observation : A W-statistic of 1251.603 and p-value of 0.00 clearly shows aggressive evidence against Homogeneity of Variance ; the data is considered not homogenous at all and structrually different.